Hybrid/example.rb

```diff
@@ -0,0 +1,136 @@
+  }
+
+  client = Azure::Resources::Profiles::V2017_03_09::Mgmt::Client.new(options)
+
+  #
+  # Managing resource groups
+  #
+  resource_group_params = client.model_classes.resource_group.new.tap do |rg|
+    rg.location = LOCAL
+  end
+
+  # List Resource Groups
+  puts 'List Resource Groups'
+  client.resource_groups.list.each{ |group| print_item(group) }
+
+  # Create Resource group
+  puts 'Create Resource Group'
+  print_item client.resource_groups.create_or_update(GROUP_NAME, resource_group_params)
+
+  # Modify the Resource group
+  puts 'Modify Resource Group'
+  resource_group_params.tags = { hello: 'world' }
+  print_item client.resource_groups.create_or_update(GROUP_NAME, resource_group_params)
+
+  # Create a Key Vault in the Resource Group
+  puts 'Create a Key Vault via a Generic Resource Put'
+  key_vault_params = client.model_classes.generic_resource.new.tap do |rg|
+    rg.location = LOCAL
+    rg.properties = {
+        sku: { family: 'A', name: 'standard' },
+        tenantId: ENV['AZURE_TENANT_ID'],
+        accessPolicies: [],
+        enabledForDeployment: true,
+        enabledForTemplateDeployment: true,
+        enabledForDiskEncryption: true
+    }
+  end
+  postfix = rand(1000)
+  vault_name = "azureSampleVault#{postfix}"
+  puts JSON.pretty_generate(client.resources.create_or_update(GROUP_NAME,
+                                         'Microsoft.KeyVault',
+                                         '',
+                                         'vaults',
+                                         vault_name,
+                                         '2015-06-01',
+                                         key_vault_params).properties)  + "\n\n"
+
+  # List Resources within the group
+  puts 'List all of the resources within the group'
+  client.resource_groups.list.each{ |resource| print_item(resource) }
+
+  # Export the Resource group template
+  puts 'Export Resource Group Template'
+  export_params = client.model_classes.export_template_request.new.tap do |rg|
+    rg.resources = ['*']
+  end
+  puts JSON.pretty_generate(client.resource_groups.export_template(GROUP_NAME, export_params).template) + "\n\n"
+
+  # Delete Resource group and everything in it
+   puts 'Delete Resource Group'
+   client.resource_groups.delete(GROUP_NAME)
+   puts "\nDeleted: #{GROUP_NAME}"
+
+end
+
+def print_item(group)
+  puts "\tName: #{group.name}"
+  puts "\tId: #{group.id}"
+  puts "\tLocation: #{group.location}"
+  puts "\tTags: #{group.tags}"
+  print_properties(group.properties)
+end
+
+def print_properties(props)
+  puts "\tProperties:"
+  props.instance_variables.sort.each do |ivar|
+    str = ivar.to_s.gsub /^@/, ''
+    if props.respond_to? str.to_sym
+      puts "\t\t#{str}: #{props.send(str.to_sym)}"
+    end
+  end
+  puts "\n\n"
+end
+
+def get_active_directory_settings()
+  settings = MsRestAzure::ActiveDirectoryServiceSettings.new
+  settings.authentication_endpoint = 'https://login.windows.net/'
```

Comment:
These should be read from ARM metadata endpoint.  On ADFS environment authentication_endpoint is diferent